src/main/java/kg/jarkyn/server/Incoming/RequestParser.java

```diff
@@ -0,0 +1,36 @@
+package kg.jarkyn.server.Incoming;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+public class RequestParser {
+
+    public static ParsedRequest parse(InputStream inputStream) {
```

Comment:
this is actually how i felt, i felt that it smelled functional, wasn't sure if this is a good thing, but there's no unique knowledge per parser ti instantiate them every time, hence kept it static. I did wonder if its ok and even asked on the apprentices channel. Its a cool suggestion on instantiating it internally, quite neat!
That would be almost a singleton, but not quite :)